### ISSUE TITLE
Separate model specific and model agnostic hyperparameters

### ISSUE THREAD

**Issue by egillax** [OHDSI/DeepPatientLevelPrediction#27]
So when users add custom models they only need to define their own model hyperparameters

**Comment by egillax** [comment]
started working on a setEstimator function:

#' setEstimator
#' 
#' @description 
#' creates settings for the Estimator, which takes a model and trains it
#' 
#' @name setEstimator
#' @param learningRate          what learning rate to use
#' @param weightDecay           what weight_decay to use
#' @param optimizer             which optimizer to use
#' @param scheduler             which learning rate scheduler to use
#' @param earlyStopping         If earlyStopping should be used which stops the training of your metric is not improving
#' @param earlyStoppingMetric   Which parameter to use for early stopping
#' @param patience              patience for earlyStopper
#' @param hyperparameterMetric  which metric to use for hyperparameter, loss, auc, auprc or a custom function

This will add more customisation to the training loop. Users can add their own optimizer or schedulers, either from torch or writing their own. Can also choose metrics to use for earlyStopping and hyperparameter-search. Should also be able to write their own functions for that.

I only wrote the documentation so far. I'll need to make sure that if you put in multiple values of these values that they are included in the hyperparameter tuning.

**Comment by egillax** [comment]
Fixed by #51
